src/playercards/AllCardsBag.ttslua

```diff
@@ -147,89 +333,216 @@ function buildSupplementalIndexes()
-            cycleName = "core"
-          end
-          if cycleIndex[cycleName] == nil then
-            cycleIndex[cycleName] = { }
+      if cardMetadata.type then
+        if not typeIndex[string.lower(cardMetadata.type)] then
+          typeIndex[string.lower(cardMetadata.type)] = { }
+        end
+        table.insert(typeIndex[string.lower(cardMetadata.type)], cardMetadata.id)
+      end
+
+      if cardMetadata.slot then
+        local slotString = cardMetadata.slot .. "|"
+        -- Getting the initial number of slots by counting the '|' characters
+        local slotCount = select(2, string.gsub(slotString, "|", ""))
+        -- Using regex to hit every slot in the slot string and add it to the corresponding subindex
+        for slot in slotString:gmatch("[^|]+") do
+          table.insert(slotIndex[string.lower(slot)], cardMetadata.id)
+          -- If the slot is a x2 slot, it will increase the slotCount, and also add it to the singular slot index (Hand for "Hand x2" for example)
+          if string.match(slot, " x2") then
+            slotCount = slotCount + 1
+            table.insert(slotIndex[string.lower(string.sub(slot, 1, -4))], cardMetadata.id)
           end
-          table.insert(cycleIndex[cycleName], cardMetadata.id)
         end
+        table.insert(slotIndex[slotCount], cardMetadata.id)
+      end
+
+      if cardMetadata.cost then
+        if not costIndex[cardMetadata.cost] then
+          costIndex[cardMetadata.cost] = { }
+        end
+        table.insert(costIndex[cardMetadata.cost], cardMetadata.id)
+      end
+
+      if cardMetadata.permanent then
+        table.insert(permanentIndex, cardMetadata.id)
+      end
+
+      if cardMetadata.uses then
+        local usesCount = cardMetadata.uses[1].count or -1
+        if not usesIndex[usesCount] then
+          usesIndex[usesCount] = { }
+        end
+        table.insert(usesIndex[usesCount], cardMetadata.id)
+        if not usesIndex[string.lower(cardMetadata.uses[1].type)] then
+          usesIndex[string.lower(cardMetadata.uses[1].type)] = { }
+        end
+        table.insert(usesIndex[string.lower(cardMetadata.uses[1].type)], cardMetadata.id)
       end
+      -- Old Taboo Code
+      -- if bonded then
+      --   for _, bondData in ipairs(bonded) do
+      --     local bondCardMetadata = cardIdIndex[bondData.id].metadata
+      --     if bondCardMetadata.uses then
+      --       if not usesIndex[bondCardMetadata.uses[1].count] then
+      --         usesIndex[bondCardMetadata.uses[1].count] = { }
+      --       end
+      --       table.insert(usesIndex[bondCardMetadata.uses[1].count], cardMetadata.id)
+      --       if not usesIndex[string.lower(bondCardMetadata.uses[1].type)] then
+      --         usesIndex[string.lower(bondCardMetadata.uses[1].type)] = { }
+      --       end
+      --       table.insert(usesIndex[string.lower(bondCardMetadata.uses[1].type)], cardMetadata.id)
+      --     end
+      --   end
+      -- end
```

Comment:
Should be removed

Reply:
Whoops yeah, good catch (Still bitter though XD)